app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -1,43 +1,85 @@
 package org.oppia.app.parser
 
 import org.oppia.app.model.Fraction
+import org.oppia.app.topic.FractionParsingErrors
 import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Integer.parseInt
+import java.util.regex.Pattern
 
 /** This class contains method that helps to parse string to fraction. */
 class StringToFractionParser {
   private val wholeNumberOnlyRegex = """^-? ?(\d+)$""".toRegex()
   private val fractionOnlyRegex = """^-? ?(\d+) ?/ ?(\d)+$""".toRegex()
   private val mixedNumberRegex = """^-? ?(\d)+ ?(\d+) ?/ ?(\d)+$""".toRegex()
+  private val invalidCharsRegex = """^[\d\s/-]+$""".toRegex()
+  private val fractionRegex = """^\s*-?\s*((\d*\s*\d+\s*\/\s*\d+)|\d+)\s*$""".toRegex()
+
+  fun fromRawInputString(inputText: String): FractionParsingErrors {
```

Comment:
Perhaps: checkForErrors would be a clearer name.

Also, please add a KDoc briefly explaining how to use this method, and when it's expected to be used.